Bio/NeuralNetwork/Gene/Signature.py

```diff
@@ -140,22 +141,10 @@ def __init__(self, signatures, max_gap):
         self._max_gap = max_gap
 
         # check to be sure the signatures are all the same size
-        # only do this if we actually have signatures
-        if len(self._signatures) > 0:
-            first_sig_size = len(self._signatures[0][0])
-            second_sig_size = len(self._signatures[0][1])
-
-            assert first_sig_size == second_sig_size, \
-                   "Ends of the signature do not match: %s" \
-                   % self._signatures[0]
-
-            for sig in self._signatures:
-                assert len(sig[0]) == first_sig_size, \
-                       "Got first part of signature %s, expected size %s" % \
-                       (sig[0], first_sig_size)
-                assert len(sig[1]) == second_sig_size, \
-                       "Got second part of signature %s, expected size %s" % \
-                       (sig[1], second_sig_size)
+        sig_sizes = set((len(sig) for sig in chain(*signatures)))
+        if len(sig_sizes) > 1:
+            raise ValueError('Inconsistent signature sizes: {sizes}'.format(
+                sizes=','.join((str(size) for size in sorted(sig_sizes)))))
```

Comment:
I think this change makes sense - is ``self._signatures`` a list of pairs of sizes?

Reply:
Yes self._signatures, or signatures is a list of pairs.